### PR TITLE
XHR/MIME Sniffing: adjust Content-Type manipulation and MIME type ser…

### PR DESCRIPTION
**Pull request by annevk**
…ialization

Tests for https://github.com/whatwg/xhr/pull/224 and https://github.com/whatwg/mimesniff/pull/86. They are combined in a single commit as otherwise there would be two rounds of changes to the XMLHttpRequest tests, while they are really supposed to be coupled.

### CODE REVIEW

File: mimesniff/mime-types/resources/mime-types.json [hunk 1]
```diff
@@ -226,11 +226,11 @@
...
+    "output": "x/x; x=\"\t !\\\"#$%&'()*+,-./0123456789:; <=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\\\]^_`abcdefghijklmnopqrstuvwxyz{|}~\u0080\u0081\u0082\u0083\u0084\u0085\u0086\u0087\u0088\u0089\u008A\u008B\u008C\u008D\u008E\u008F\u0090\u0091\u0092\u0093\u0094\u0095\u0096\u0097\u0098\u0099\u009A\u009B\u009C\u009D\u009E\u009F\u00A0\u00A1\u00A2\u00A3\u00A4\u00A5\u00A6\u00A7\u00A8\u00A9\u00AA\u00AB\u00AC\u00AD\u00AE\u00AF\u00B0\u00B1\u00B2\u00B3\u00B4\u00B5\u00B6\u00B7\u00B8\u00B9\u00BA\u00BB\u00BC\u00BD\u00BE\u00BF\u00C0\u00C1\u00C2\u00C3\u00C4\u00C5\u00C6\u00C7\u00C8\u00C9\u00CA\u00CB\u00CC\u00CD\u00CE\u00CF\u00D0\u00D1\u00D2\u00D3\u00D4\u00D5\u00D6\u00D7\u00D8\u00D9\u00DA\u00DB\u00DC\u00DD\u00DE\u00DF\u00E0\u00E1\u00E2\u00E3\u00E4\u00E5\u00E6\u00E7\u00E8\u00E9\u00EA\u00EB\u00EC\u00ED\u00EE\u00EF\u00F0\u00F1\u00F2\u00F3\u00F4\u00F5\u00F6\u00F7\u00F8\u00F9\u00FA\u00FB\u00FC\u00FD\u00FE\u00FF\""
```
Comment (domenic) [inline mimesniff/mime-types/resources/mime-types.json]: The second semicolon here should not have a space after it.

**Review by domenic — CHANGES_REQUESTED** [review]

File: mimesniff/mime-types/resources/mime-types.json (hunk 1 above)
Comment (annevk) [reply mimesniff/mime-types/resources/mime-types.json]: Thanks, fixed!

**Comment by annevk** [comment]
Review ping! (Ta!)

**Review by wisniewskit — APPROVED** [review]
LGTM with domenic's suggested fix.

**Review by domenic — APPROVED** [review]

**Review by domenic — CHANGES_REQUESTED** [review]
Marking as "request changes" to remind us to update data-urls.json too.

**Comment by annevk** [comment]
So `data:` URLs have that `text/plain;charset=US-ASCII` default. That should probably remain without space. There's examples of both kinds here and whenever we don't serialize we should probably not alter. I really hope we don't find more breakage meaning we effectively always have to scan or preserve whitespace somehow.

**Comment by annevk** [comment]
Sigh, this is wrong: https://fetch.spec.whatwg.org/#data-url-processor. It'd have to be `text/plain; charset=US-ASCII` (with space) as it's stored as a record. Maybe we should not add the space after all.

**Comment by annevk** [comment]
I created #13779 to supersede this PR with a much more limited scope (no longer changing the serialization of all MIME types).

**Pull request closed without merging**